src/dtgtk/thumbnail.c

```diff
@@ -1224,17 +1225,22 @@ static void _thumb_resize_overlays(dt_thumbnail_t *thumb)
   {
     gtk_widget_get_size_request(thumb->w_main, &width, &height);
 
+    const int margin_l = MAX(0, thumb->img_margin->left * width / 100);
+    const int margin_t = MAX(0, thumb->img_margin->top * height / 100);
+    const int margin_r = MAX(0, thumb->img_margin->right * width / 100);
+    const int margin_b = MAX(0, thumb->img_margin->bottom * height / 100);
+
     // we need to squeeze 5 stars + 1 reject + 1 colorlabels symbols on a thumbnail width
     // stars + reject having a width of 2 * r1 and spaced by r1 => 18 * r1
     // colorlabels => 3 * r1 + space r1
-    // inner margins are 0.02 * width
+    // inner margins are defined in css (margin_* values)
 
     // retrieves the size of the main icons in the top panel, thumbtable overlays shall not exceed that
-    const float r1 = fminf(max_size / 2.0f, 0.94 * width / 22.0f);
+    const float r1 = fminf(max_size / 2.0f, (width - margin_l - margin_r) / 22.0f);
 
     // file extension
-    gtk_widget_set_margin_top(thumb->w_ext, 0.03 * width);
-    gtk_widget_set_margin_start(thumb->w_ext, 0.03 * width);
+    gtk_widget_set_margin_top(thumb->w_ext, margin_t);
```

Comment:
it's not the same thing though. all margins were defined to 3% of the width, now it's 3% of the width for horizontal margins or 3% of the height for vertical ones. They can be different, so you will generally not be able to get an homogenous margin in thumbnail.

Reply:
well homogeneous margins are the default in css for this reason. Now if user decide to use smaller/greater margins, I think we need to follow that settings for icons too.
Of course if an user decide to have different margins for left and top, he'll get not homogeneous margins, but that's what he has decide !

Reply:
I agree with @AlicVB : If we set default same margins on CSS, I'm also more to not hardcode them. The user who want to tweak that do what he want and if he complain about not homogenous margins, it's his problem. Most important thing is that we, developers team, deliver a default homogenous margins.